### ISSUE TITLE
Structured Syntax Suffix support?

### ISSUE THREAD

**Issue by dkubb** [zmthy/http-media#15]
Are there any plans to support the Structured Syntax Suffix RFC: http://trac.tools.ietf.org/html/draft-ietf-appsawg-media-type-suffix-regs-02

Specifically I would want to be able to use the `+json` suffix as defined in http://trac.tools.ietf.org/html/draft-ietf-appsawg-media-type-suffix-regs-02#section-3.1 with a wildcard like the following:

```
application/*+json
```

It would also be nice to be able to even be able to specify:

```
*/*+json
```

**Comment by zmthy** [comment]
No, but I'd welcome a pull request. Can you provide a use case for this functionality?

**Comment by dkubb** [comment]
In an API that I work on, we can accept a valid JSON request whether it's `application/json` or a vendor tag like `application/vnd.something+json`. The `+json` suffix says that the media follows the same parsing rules as `application/json`. Granted we do currently just hard-code the specific media types, but it would be nice to be able to specify a wildcard instead.
